anvio/scgsdatabase.py

```diff
@@ -0,0 +1,791 @@
+
+    def diamonddb(self, pathgdb):
+
+        diamond = Diamond()
+        diamond.target_fasta = pathgene
+        run.info("diamonddb_stdin", pathgene)
+        diamond.makedb(pathgdb)
+
+
+    def diamondblast_stdin(self,sequencetoblast, pathgene):
+
+        diamond = Diamond()
+        diamond.target_fasta = pathgene
+        run.info("diamonddb_stdin", pathgene)
+        diamond.makedb_stdin(sequencetoblast)
+
+
+
+    def diamondblast_stdou(self,pathquery,pathdb, max_target_seqs=20, evalue=1e-05, min_pct_id=90):
+        pathdb=pathdb+".dmnd"
+        diamond = Diamond(pathdb,run=self.run, progress=self.progress, num_threads=self.num_threads)
+        #diamond = Diamond(pathdb)#, run=run_quiet, progress=progress_quiet
+        diamond.max_target_seqs = max_target_seqs
+        diamond.query_fasta=pathquery
+        run.info("diamondblast_stdou", pathdb)
+        diamond_output = diamond.blastp_stdout()
+        str_diamond_output=diamond_output.decode("utf-8")
+        return(str_diamond_output)
+
+
+
+
+
+    def refundgenes(self,pathgenes,pathrefundgenes,path_diamondb):
+        fasta=u.ReadFasta(pathgenes)
+        i=0
+        sequencetoblast=""
+        while i < len(fasta.ids):
+            if fasta.sequences[i].count('-') < len(fasta.sequences[i])*0.5:
+                sequencetoblast=sequencetoblast+">"+fasta.ids[i]+"\n"+fasta.sequences[i]+"\n"
+            i+=1
+        return(sequencetoblast)
+
+
+
+
+    def trie_blast(self,outpath,dictionnary_corres):
+        """ return dictonnary with the name of Genes from given sequence corresponding with the one from anvi'o"""
+
+        pathblastfile=os.path.join(outpath,blastfile)
+        if os.stat(pathblastfile).st_size == 0:
+            os.remove(pathblastfile)
+        hmm_gene, bacgene=self.corre(pathblastfile)
+        dictionnary_corres[bacgene]=hmm_gene
+        return dictionnary_corres;
+
+
+
+    def cleandir(self,dir):
+        shutil.rmtree(dir)
+
+    def dictionnary_level(self):
+        dictionnary_level= {}
+        dictionnary_level["domain"]=1
+        dictionnary_level["phylum"]=2
+        dictionnary_level["class"]=3
+        dictionnary_level["order"]=4
+        dictionnary_level["family"]=5
+        dictionnary_level["genus"]=6
+        dictionnary_level["species"]=7
+        return(dictionnary_level)
+
+    def make_dictionnary_level(self,taxofiles):
+        with open(taxofiles, 'r') as taxotsv:
+            linestaxo = taxotsv.readlines()
+            dictionnary_level = {}
+            for line in linestaxo:
+                names = line.split('\t')
+                code = names[0]
+                leveltaxos = names[1].split(';')
+                for leveltaxo in leveltaxos[::-1]:
+                    leveltaxo1 = re.sub(r'sp[1-9]*', '', leveltaxo)
+                    leveltaxo3 = re.sub(r'(\\*[a-z])\_[A-Z]', r'\1', leveltaxo1)
+                    leveltaxo2 = leveltaxo3.rstrip()
+                    leveltaxo = re.sub(r'\s+', r'_', leveltaxo2)
```

Comment:
this is unacceptable. `leveltaxo*` variables are killing me right now :) please find a better way to parse them. probably everything in this function can be dune in a single line of proper Python code. take it as a challenge and enjoy your creation once you are done.